### ISSUE TITLE
[FR] Support URL signatures

### ISSUE THREAD

**Issue by mattstein** [spacecatninja/craft-imager-x-aws-serverless-transformer#6]
Hi!

Currently there’s no built-in support for signed URLs that prevent unknown villains from generating their own transforms. While this isn’t a huge deal since source material is limited to what’s in predefined buckets, it’d be nice to take advantage of URL signatures since they’re supported by the Serverless Image Handler itself.

I hacked a `$signatureKey` into the Settings model and added a condition to the end of `AwsServerless::getTransformedImage()` that works great:

```php
// ... no other code harmed

if (isset($settings->signatureKey)) {
    // Prepare the signature if the settings model has a key we should use
    $signature = hash_hmac('sha256', '/' . base64_encode($encodedRequestParams), $settings->signatureKey);
    $url .= '?signature=' . $signature;
}

return new AwsServerlessTransformedImageModel($url, $image, $requestParams);
```

Happy to PR that if you’d like!

**Comment by aelvan** [comment]
A PR would be great! :)

**Comment by aelvan** [comment]
Merged PR into 2.1.0, thanks Matt!
